### PR TITLE
Report facility status from facility details

### PR DESCRIPTION
**Pull request by TaiWilkin**
## Overview

Allows users to report a facility as closed or reopened through a link in the facility details. 
Shows a ribbon in the facilities list and the facility details if a facility is closed, and shows a pending ribbon in the details. 
Shows a list of pending and confirmed status change requests. 

Connects #1232 

## Demo

Closed
Screen Shot 2021-02-15 at 8 45 06 AM
Screen Shot 2021-02-15 at 8 45 19 AM

Pending
Screen Shot 2021-02-15 at 8 45 43 AM

Status List
Screen Shot 2021-02-15 at 8 45 37 AM

Success Ribbon
Screen Shot 2021-02-15 at 8 46 19 AM

## Testing Instructions

* Run `vagrant ssh` and `./scripts/server`
* Navigate to a facility
    * You should not see a ‘report facility’ button
* Login as c8@example.com and navigate to a facility
    * You should see a ‘Report Facility as closed’ button
* Press ‘report facility as closed’
    * You should see a dialog
* Press ‘report’ without entering any text
    * The report should not submit
* Enter text and press ‘cancel’, then reopen the dialog
    * The dialog textbox should be cleared after cancelling
* Enter text and press ‘report’
    * The dialog should close
    * There should be a success alert
    * There should be a ‘pending’ ribbon in the facility details
    * There should be a ‘Status’ list with a ‘reported as closed’ item only
* Return to the facility list
    * The facility you reported should not have a ribbon on the list
* Report an additional facility
* Logout and login as c1@example.com
* Navigate to http://localhost:6543/dashboard/activityreports
    * There should be pending requests to close the facilities you reported
* Confirm one report and reject the other.
* Return to the facility list. 
    * The confirmed facility should have a closed ribbon. 
    * The rejected facility should have no ribbon.
* Navigate to the rejected facility.
    * It should have no ribbon and no Status items.
* Navigate to the confirmed facility. 
    * It should have a closed ribbon and two status items - request and verified.
    * The button should say, ‘Report facility as reopened’
* Attempt to report the facility as reopened.
    * You should get an error message about not having a contributor.
* Log out and log back in as c8@example.com, then again attempt to reopen the facility.
    * It should succeed.
    * There should be a pending ribbon. 
    * The new status item should be at the top of the Statuses list.

## Checklist

- [x] `fixup!` commits have been squashed
- [ ] CI passes after rebase
- [x] CHANGELOG.md updated with summary of features or fixes, following [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) guidelines

### CODE REVIEW

**Comment by jwalgran** [comment]
Looking at this now.

File: src/app/src/components/DashboardActivityReportToast.jsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+import Snackbar from '@material-ui/core/Snackbar';
+import Paper from '@material-ui/core/Paper';
+import Typography from '@material-ui/core/Typography';
+import IconButton from '@material-ui/core/IconButton';
+import CloseIcon from '@material-ui/icons/Close';
+
+const gradient = 'linear-gradient(-90deg, rgb(224, 32, 32) 0%, rgb(163, 79, 165) 21%, rgb(0, 145, 255) 51%, rgb(91, 201, 238) 79%, rgb(109, 212, 0) 100%)';
+
+const styles = {
```
Comment (jwalgran) [inline src/app/src/components/DashboardActivityReportToast.jsx:12]: At narrower widths the styling on the Snackbar makes it expand horizontally leaving the little rainbow bar looking out of place

Screen Shot 2021-02-15 at 12 16 24 PM

**Review by jwalgran — COMMENTED** [review]
Excellent implementation of the workflow. It worked perfectly in my testing. I made a couple of observations inline and have another sub-feature to suggest that was not in the scope of the original issue. If you expect the implementation to be too far outside of the original estimate we can open it as a separate issue.

The "Claim this facility" link prompts for a login if needed. It makes sense for us to do the same with activity reports, but the workflow is a little different since claims have a dedicated URL to navigate to and activity reporting is just done in a dialog. The simplest solution that comes to show a similarly styled dialog.

Screen Shot 2021-02-15 at 12 39 03 PM

File: src/app/src/components/FacilityDetailStatusList.jsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                            status,
+                        } = report;
+                        const state = closureState.toLowerCase();
+                        return (
+                            <ul key={report.id} style={styles.list}>
+                                <li>Reported {state} by {user} on {formatDate(created)}</li>
+                                {status === 'CONFIRMED' && (
+                                    <li>
+                                        Verified as {state} by{' '}
+                                        {statusChangeBy} on {formatDate(statusChangeDate)}
```
[the remark this answers is not in the document]
Comment (TaiWilkin) [reply src/app/src/components/FacilityDetailStatusList.jsx]: Good plan. I've updated tp drop the email:
Screen Shot 2021-02-16 at 11 05 19 AM

File: src/app/src/components/DashboardActivityReportToast.jsx (hunk 1 above)
Comment (TaiWilkin) [reply src/app/src/components/DashboardActivityReportToast.jsx:12]: Updated styling to fix:
Screen Shot 2021-02-16 at 11 04 26 AM

**Comment by TaiWilkin** [comment]
Switched to using a login prompt instead of hiding the button as suggested:
Screen Shot 2021-02-16 at 11 23 49 AM
This should be ready for another look.

**Review by jwalgran — APPROVED** [review]
Excellent fixups. 👍

**Pull request merged**